src/worker/core/player/shouldRetire.ts

```diff
@@ -9,13 +9,39 @@ import type {
 const shouldRetire = (
 	p: Player<MinimalPlayerRatings> | PlayerWithoutKey<MinimalPlayerRatings>,
 ): boolean => {
-	const age = g.get("season") - p.born.year;
-
+  
+	const season = g.get("season");
+	const originalSeason = g.get("startingSeason");
 	const forceRetireAge = g.get("forceRetireAge");
+	const forceRetireSeason = g.get("forceRetireSeason");
+
+	const age = season - p.born.year;
+
 	if (forceRetireAge >= g.get("draftAges")[1] && age >= forceRetireAge) {
 		return true;
 	}
 
+	//check how many unique seasons the player has played in
+	let uniqueSeasons = 0;
+	const seasonArray = new Array(season).fill(0);
```

Comment:
This initializes an array with `season` elements. So if it's the year 2000, it's a size 2000 array! Not the end of the world, but a bit wasteful. But if someone edited their league to start in the very distant future it would be bad :)

Reply:
haha you're right, bit of an oversight. fixed it now